sql/core/src/main/scala/org/apache/spark/sql/execution/command/views.scala

```diff
@@ -128,6 +129,15 @@ case class CreateViewCommand(
     qe.assertAnalyzed()
     val analyzedPlan = qe.analyzed
 
+    // CREATE VIEW AS INSERT INTO ... is not allowed, we should throw an AnalysisException.
+    analyzedPlan match {
+      case i: InsertIntoHadoopFsRelationCommand =>
```

Comment:
The sql parser only allows `CREATE VIEW AS query` here, a query can only be a `SELECT ...` or `INSERT INTO ...` or a CTE, so perhaps we don't have to consider other commands here.